definitions/npm/react-navigation_v3.x.x/flow_v0.60.x-/react-navigation_v3.x.x.js

```diff
@@ -1231,6 +1232,7 @@ declare module 'react-navigation' {
     }) => void,
     getTestIDProps: (scene: TabScene) => (scene: TabScene) => any,
     renderIcon: (scene: TabScene) => React$Node,
+    getButtonComponent: (scene: TabScene) => React$Node,
```

Comment:
Why one is returning a React.Element and the other a React.Node, can you point us to some code/docs?

Reply:
A quick search through the react-navigation github org shows it should be more `React.ComponentType<*>`

Reply:
Both `React$Node` & `React$Element` are acceptable as React Component return type
This `React.ComponentType` might be the newer type declaration

Reply:
They are three distinct types serving different purposes.  
From what I understand `getButtonComponent` is not rendering anything, it returns a component. 

Again, please point us to some docs or material asserting those changes.

Reply:
I looked into this (sorry for the delay). `_TabBarTopProps` does not actually even have a `getButtonComponent` prop ([source](https://github.com/react-navigation/react-navigation-tabs/blob/master/src/views/MaterialTopTabBar.js)), whereas `_TabBarBottomProps` does and it is a `React.ComponentType` as you said ([source](https://github.com/react-navigation/react-navigation-tabs/blob/master/src/views/BottomTabBar.js)).

I think we can probably just close this PR - I'll put up a new one in a sec.